client/src/widgets/CodeChecker.vue

```diff
@@ -0,0 +1,274 @@
+<template>
+  <div class="codeChecker">
+    <MainInput
+      :name="'code'"
+      :type="'text'"
+      :placeholder="'Code from view special board ...'"
+    />
+    <div class="codeChecker__buttons">
+      <ActionButton
+        :icon="'check'"
+        :is-green="true"
+        class="codeChecker__btn"
+        @click-action="checkCode"
+        >Check</ActionButton
+      >
+      <ActionButton
+        id="open-modal"
+        :icon="'code'"
+        class="codeChecker__btn"
+        @click-action="startScan"
+        >Scan QR code</ActionButton
+      >
+    </div>
+    <IonModal
+      ref="modal"
+      trigger="open-modal"
+      class="codeChecker__modal"
+      @didDismiss="stopCamera"
+    >
+      <div v-show="!isLoaded" class="codeChecker__loader"></div>
+      <div v-show="isLoaded">
+        <video ref="videoElement" width="110%" :hidden="!scanActive"></video>
+      </div>
+    </IonModal>
+    <canvas ref="canvasElement" hidden></canvas>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { IonModal } from '@ionic/vue';
+import MainInput from '@/components/inputs/MainInput.vue';
+import ActionButton from '@/components/buttons/ActionButton.vue';
+import { forwardAnimation } from '@/animations/navigateAnimations';
+import { object, string } from 'yup';
+import { useIonRouter } from '@ionic/vue';
```

Comment:
dublikat importu , tam wyzej jest import z tego samego miejsca to mozna w jeden import dac